### PR TITLE
Stop Card Header

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR reuses the stop card header portion of the nearby view into the stop schedule viewer.
This ensures consistent display of stop names and operator logos. There are slight visual changes in the stop schedule viewer header content as a result (see screenshots).

## PR Checklist
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

| Before | After |
|--------|-------|
|  Before screenshot  |  After screenshot   |

### CODE REVIEW

File: lib/components/util/link.tsx [hunk 1]
```diff
@@ -31,4 +31,9 @@ const Link = ({
   </RouterLink>
 )
 
-export default Link
+const DummyLink = ({ children, to }: Props): JSX.Element => (
+  // Do not render router links in snapshots.
```
Comment (amy-corson-ibigroup) [inline lib/components/util/link.tsx:35]: Why do we want to avoid this?

File: lib/components/viewers/stop-schedule-viewer.tsx [hunk 2]
```diff
@@ -84,6 +81,46 @@ const StyledAlert = styled(Alert)`
...
+    margin: 25px 0 0;
+  }
+
+  input[type='date'] {
+    background: inherit;
+    border: none;
+    clear: right;
+    cursor: pointer;
+    outline: none;
+    width: 125px;
```
Comment (amy-corson-ibigroup) [inline lib/components/viewers/stop-schedule-viewer.tsx:99]: Is there a reason this width is fixed? It looks like it's causing a little gaping between the icon and the input on Chrome.
image

File: lib/components/viewers/nearby/stop.tsx [hunk 3]
```diff
@@ -132,61 +78,21 @@ const Stop = ({
...
-          >
-            <IconWithText Icon={Calendar}>
-              <FormattedMessage id="components.StopViewer.viewSchedule" />
-            </IconWithText>
-          </button>
-        </div>
-        {fromToSlot}
-      </CardBody>
+    <Card>
+      <StopCardHeader
```
Comment (amy-corson-ibigroup) [inline lib/components/viewers/nearby/stop.tsx:85]: Thoughts on adding a `titleAs="h4"` here?

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks great! Just a few comments and questions

File: lib/components/util/link.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/util/link.tsx:35]: Yeah I need to update the comment. you would have to setup an entire stack of providers, routes, and switches in the tests, and I didn't find an easy way to do it.

File: lib/components/viewers/stop-schedule-viewer.tsx (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/components/viewers/stop-schedule-viewer.tsx:99]: In Firefox, the text field takes up more space.

File: lib/components/viewers/nearby/stop.tsx (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/components/viewers/nearby/stop.tsx:85]: I think each card in the nearby view is part of a list already, so there is no need for headers there.

File: lib/components/util/link.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/util/link.tsx:35]: The alternative otherwise is to just render an anchor `<a>` with the desired `href`.

Comment (binh-dam-ibigroup) [reply lib/components/util/link.tsx:35]: Thoughts, @miles-grant-ibigroup?

Comment (miles-grant-ibigroup) [reply lib/components/util/link.tsx:35]: Why are we not using anchor tags?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Thanks for all this cleanup. Nothing jumping out at me in the code, but I'm observing that the schedule viewer (what used to be the stop viewer) is not showing multiple operator logos as the nearby view is. Also, what happened to the operator fallback icons? I'm seeing just a title on stops with operators that have no logos

**Comment by binh-dam-ibigroup** [comment]
> I'm observing that the schedule viewer (what used to be the stop viewer) is not showing multiple operator logos as the nearby view is.

See the "After" screenshot for multiple operator logos. Is there another case for multiple operators I missed?

**Comment by binh-dam-ibigroup** [comment]
> Also, what happened to the operator fallback icons? I'm seeing just a title on stops with operators that have no logos

Good point, that was not supported in the nearby view cards, I'll look into adding those back.

**Comment by binh-dam-ibigroup** [comment]
> > Also, what happened to the operator fallback icons? I'm seeing just a title on stops with operators that have no logos
> 
> Good point, that was not supported in the nearby view cards, I'll look into adding those back.

For stops with operators without a logo, that was already supported by the nearby view card, so support carries over to the schedule viewer, with an adjustment to the image height in https://github.com/opentripplanner/otp-react-redux/pull/1169/commits/9ea8750deed8ceb8dcbdfbf87fbab452803bcde5 (see screenshot).

image

**Comment by miles-grant-ibigroup** [comment]
> > > Also, what happened to the operator fallback icons? I'm seeing just a title on stops with operators that have no logos
> > 
> > 
> > Good point, that was not supported in the nearby view cards, I'll look into adding those back.
> 
> For stops with operators without a logo, that was already supported by the nearby view card, so support carries over to the schedule viewer, with an adjustment to the image height in [9ea8750](https://github.com/opentripplanner/otp-react-redux/commit/9ea8750deed8ceb8dcbdfbf87fbab452803bcde5) (see screenshot).
> 
> image

Wait I think we decided in a design meeting that we didn't want the pin to show if operator logos were present

**Comment by binh-dam-ibigroup** [comment]
> Wait I think we decided in a design meeting that we didn't want the pin to show if operator logos were present

Fixed in https://github.com/opentripplanner/otp-react-redux/pull/1169/commits/ebdbd0fe6b0d6186569b27990c7c4d6bcda8c1ed.

**Comment by miles-grant-ibigroup** [comment]
Not sure why Percy is failing

**Comment by binh-dam-ibigroup** [comment]
> Not sure why Percy is failing

Percy is failing because I pasted new mock responses with more recent schedules.

**Comment by binh-dam-ibigroup** [comment]
> Not sure why Percy is failing

Okay, turns out there was more intricate logic issues that I dealt with in https://github.com/opentripplanner/otp-react-redux/pull/1169/commits/5de6235ad6e43a354f8c37bcd7744db718c5d216.

**Comment by binh-dam-ibigroup** [comment]
@amy-corson-ibigroup and @miles-grant-ibigroup Any last thoughts on recent changes? If not I will merge this tomorrow.

**Comment by miles-grant-ibigroup** [comment]
Changes make sense to me! There's some small cleanup nitpicks but nothing that can't be fixed later

**Pull request merged**